app/Http/Controllers/ProfileController.php

```diff
@@ -135,4 +135,95 @@ public function mobileCarrier(Request $request){
             return (new ErrorResource($res))->response()->setStatusCode(400);
         }
     }
+
+    public function updateProfile(Request $request, Validate $validate){ 
+      
+       $user = $request->user();
+       $input = $request->all();
+       $validationErrors = $validate->validate($request, $this->rules->getUpdateProfileValidatonRules(),[]);
+       if( $validationErrors ){
+           return (new ErrorResource($validationErrors))->response()->setStatusCode(400);
+       }
+
+       try{           
+            $private_flags = [];
```

Comment:
Multiple if's are considered as bad design. We can do something similar to this

```php

$fields = ['first_name', 'middle_name', 'last_name', 'address_1', 'address_2', 'city', 'state', 'country', 'postal_code', 'gender', 'birthday', 'gender', 'language', 'phone_number', 'mobile_carrier',  'default_algo'];

foreach ($fields as $f) {
    if(isset($input[$f])) {
        $user->$f = $input[$f];
    }
}

```

Or we can create a `key value` array. I also need to think on this.  But the crux is we need to avoid multiple if's for sure